jupyterhub/orm.py

```diff
@@ -939,6 +942,17 @@ def ping_connection(connection, branch=None):
             # restore "close with result"
             connection.should_close_with_result = save_should_close_with_result
 
+    # sqla v1/v2 compatible invocation of @event.listens_for:
+    def ping_connection_v1(connection, branch=None):
+        """sqlalchemy < 2.0 compatibility"""
+        return ping_connection(connection)
+
+    if int(sqlalchemy.__version__.split(".", 1)[0]) >= 2:
+        listener = ping_connection
+    else:
+        listener = ping_connection_v1
+    event.listens_for(engine, "engine_connect")(listener)
```

Comment:
I think the assignment would get removed by the linter, since it's unused.

It's a bit odd that sqlalchemy does this with decorators, since the after-decoration function isn't ever used, it's really a pure 'registration' function with no meaningful return value, other than to satisfy decorator syntax.